### ISSUE TITLE
FormCheckbox old value

### ISSUE THREAD

**Issue by Meffistroler** [protonemedia/laravel-form-components#52]
![image](https://user-images.githubusercontent.com/36982709/114143654-1461ff00-991d-11eb-9042-6c5a1a439d09.png)

ProtoneMedia\LaravelFormComponents\Components\FormCheckbox.
Starts on line 45
``
...
if (is_array($boundValue)) {
        $this->checked = in_array($value, $boundValue);
        return;
    }
...
``
What about collection support?

**Comment by kb6673** [comment]
I think I've had the same issue. Is this when the checkbox is checked or unchecked

**Comment by Meffistroler** [comment]
> I think I've had the same issue. Is this when the checkbox is checked or unchecked?

When a checkbox is checked and that position depends on multiple values.

**Comment by hiayongkuy** [comment]
I have faced this issue as well when using @bind functionality with [spatie
/
laravel-permission](https://github.com/spatie/laravel-permission) $user->roles Relationship 

As I reviewed the code of HandlesBoundValues trait, i found out that  the trait does not handle MorphToMany relationship which return from $user->roles() QueryBuilder.

```
private function getAttachedKeysFromRelation($bind, string $name): ?array
    {
        if (!$bind instanceof Model) {
            return data_get($bind, $name);
        }

        $relation = $bind->{$name}();

        if ($relation instanceof BelongsToMany) {
            $relatedKeyName = $relation->getRelatedKeyName();

            return $relation->getBaseQuery()
                ->get($relation->getRelated()->qualifyColumn($relatedKeyName))
                ->pluck($relatedKeyName)
                ->all();
        }

        if ($relation instanceof MorphMany) {
            $parentKeyName = $relation->getLocalKeyName();

            return $relation->getBaseQuery()
                ->get($relation->getQuery()->qualifyColumn($parentKeyName))
                ->pluck($parentKeyName)
                ->all();
        }

        return data_get($bind, $name);
    }
```

**Comment by pascalbaljet** [comment]
Fixed in v3.1.0.
